File: src/types/task.ts
// 任务状态枚举
export type TaskStatus = 'todo' | 'in_progress' | 'completed' | 'blocked'

// 任务优先级
export type TaskPriority = 'low' | 'medium' | 'high' | 'urgent'

// 任务接口定义
export interface Task {
  id: string
  title: string
  description: string
  status: TaskStatus
  priority: TaskPriority
  assignee: string
  startDate?: string
  dueDate?: string
  progress: number // 0-100
  tags: string[]
  createdAt: string
  updatedAt: string
}

// 任务看板列定义
export interface BoardColumn {
  id: TaskStatus
  title: string
  color: string
}

// 进度统计
export interface ProgressStats {
  totalTasks: number
  completedTasks: number
  inProgressTasks: number
  blockedTasks: number
  completionRate: number
}

// 用户接口
export interface User {
  id: string
  name: string
  avatar?: string
  role: string
}

File: backend/src/main/java/com/career/plan/entity/Task.java
package com.career.plan.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private String status;  // TODO, IN_PROGRESS, DONE
    
    private Integer progress;  // 0-100
    
    private String priority;  // LOW, MEDIUM, HIGH
    
    @Column(name = "assigned_to")
    private String assignedTo;
    
    @Column(name = "due_date")
    private LocalDateTime dueDate;
    
    private Boolean confirmed = false;
    
    @Column(name = "confirmed_at")
    private LocalDateTime confirmedAt;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
